department-of-veterans-affairs/caseflow
# Reader Document List: Change popup menus to work with CSS instead of JS

This is trying to fix: https://sentry.ds.va.gov/department-of-veterans-affairs/caseflow/issues/1277/

### Description

Somewhat of a shot in the dark, as it seems to only happen on IE and I can't get my virtual machine running IE to work with caseflow. However, the call stack gives us two hints: `setFilterIconPosition` is in the call stack, and the error is: `Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.`

Looks like a stack overflow exception. Looking at this code path we're doing some weird things to position a pop-up menu using JS rather than natively with CSS. It seems like we reposition this menu in componentDidUpdate. This could cause an infinitely loop of adjusting the position. This change tries to fix this by natively implementing this in CSS instead of JS! At the very least it reduces complexity.

![categorydropdown](https://user-images.githubusercontent.com/3885236/39783657-c4a2091a-52e3-11e8-80e3-4fcc138bba9e.gif)


### Acceptance Criteria 
- [x] Tag and Category dropdowns both still work and are placed correctly, even when the window is resized.

### Testing Plan
1. Go to a document list in Reader
1. Try to filter on categories and tags.
1. Ensure the popup menu is still placed correctly.
1. Try toggling options and ensure they still work.
